lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -40,9 +40,8 @@ def create_vm(attrs)
 
             attrs[:comment] ||= ""
             attrs[:quota] = attrs[:quota].present? ? client.system_service.data_centers_service.data_center_service(datacenter).quotas_service.quota_service(attrs[:quota]).get : nil
-
-            if attrs[:cores].present?
-              cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores], :sockets => "1")
+            if attrs[:cores].present? || attrs[:sockets].present?
+              cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores] || "1", :sockets => attrs[:sockets] || "1")
```

Comment:
What will happen if I have defined 2 sockets and no cores?

Reply:
Also you can use [#fetch](https://ruby-doc.org/core-2.2.0/Hash.html#method-i-fetch) method to simplify the code:

``` ruby
cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs.fetch(:cores, "1"), :sockets => attrs.fetch(:sockets, "1"))
```

Reply:
if there are 2 sockets and no cores then the core amount should be 1.